### PR TITLE
Functional tests - Add test 'Sort contacts'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add functional test `Sort contacts`
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="functional/BO/13_shopParameters/05_contact/04_sortContacts" URL_FO=shopURL/ npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17741)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/05_contact/04_sortContacts.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  before(async function () {
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    this.pageObjects = await init();
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+  });
+
+  // Login into BO and go to contact page
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/05_contact/04_sortContacts.js:42]: Can you describe what does the test do exactly, please ?

Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/05_contact/04_sortContacts.js:42]: Done 
I added description to first describe and a comment before tests array

**Review by SimonGrn — DISMISSED** [review]

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/05_contact/04_sortContacts.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  tests.forEach((test) => {
+    it(`should sort by '${test.args.sortBy}' '${test.args.sortDirection}' And check result`, async function () {
+      await testContext.addContextItem(this, 'testIdentifier', test.args.testIdentifier, baseContext);
+      let nonSortedTable = await this.pageObjects.contactsPage.getAllRowsColumnContent(test.args.sortBy);
+      await this.pageObjects.contactsPage.sortTable(test.args.sortBy, test.args.sortDirection);
+      let sortedTable = await this.pageObjects.contactsPage.getAllRowsColumnContent(test.args.sortBy);
+      if (test.args.isFloat) {
+        nonSortedTable = await nonSortedTable.map(text => parseFloat(text, 10));
+        sortedTable = await sortedTable.map(text => parseFloat(text, 10));
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/05_contact/04_sortContacts.js]: `parseFloat` doesn't have two parameters : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat

Comment (SimonGrn) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/05_contact/04_sortContacts.js]: Damn, I thought it did, but apparently that's just `parseInt` :o

Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/05_contact/04_sortContacts.js]: I thought that too, so I deleted the second parameter in all sort tests. thanks @Progi1984

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**